saritasa_s3_tools/django/serializers.py

```diff
@@ -0,0 +1,109 @@
+import typing
+
+import humanize
+from django.conf import settings
+from rest_framework import exceptions, request, serializers
+
+from .. import configs, constants
+from . import fields
+
+
+class S3RequestParamsSerializer(serializers.Serializer):
+    """Serializer for validation s3 uploading fields."""
+
+    config = fields.S3FileTypeConfig()
+    filename = serializers.CharField()
+    content_type = serializers.CharField()
+    content_length = serializers.IntegerField()
+
+    def __init__(
+        self,
+        context_request: request.Request | None = None,
+        *args,  # noqa: ANN002
+        **kwargs,
+    ) -> None:
+        """Set current user."""
+        super().__init__(*args, **kwargs)
+        self._request: request.Request | None = context_request
+        self._user = getattr(self._request, "user", None)
+
+    def validate_config(
+        self,
+        config: configs.S3FileTypeConfig,
+    ) -> configs.S3FileTypeConfig:
+        """Check that user can use dest."""
+        if config.auth and not config.auth(self._user):
+            raise exceptions.ValidationError(
+                "Current user can't use this destination",
+            )
+        return config
+
+    def validate(self, attrs: dict[str, typing.Any]) -> dict[str, typing.Any]:
+        """Perform validations.
+
+        Check what input data is valid for specified configuration.
+
+        """
+        errors: dict[str, str] = {}
+        config: configs.S3FileTypeConfig = attrs["config"]
+        filename: str = attrs["filename"]
+        content_type: str = attrs["content_type"]
+        content_length: int = attrs["content_length"]
```

Comment:
Can a `KetError` occur here?

Reply:
Hmm shouldn't be possible since all fields are required